libethcore/Transaction.cpp

```diff
@@ -147,3 +156,29 @@ bigint TransactionBase::gasRequired(bool _contractCreation, bytesConstRef _data,
 	return ret;
 }
 
+h256 TransactionBase::sha3(IncludeSignature _sig) const
+{
+	if (_sig == WithSignature && m_hashWith)
+		return m_hashWith;
+
+	RLPStream s;
+	if (m_chainId == 1 && m_type != NullTransaction) //EIP155 is On
```

Comment:
I suggest to make it useful for any valid chain id. So it should be `m_chainId > 0`.